module-system/module-implementations/sov-evm/src/evm/conversions.rs

```diff
@@ -116,64 +114,58 @@ impl From<EvmTransaction> for Transaction {
             other: OtherFields::default(),
-        }
+        })
     }
 }
 
-use reth_primitives::{Bytes as RethBytes, TransactionSigned};
+#[derive(Error, Debug)]
+pub enum RawEvmTxConversionError {
+    #[error("Empty raw transaction data")]
+    EmptyRawTransactionData,
+    #[error("Failed to decode signed transaction")]
+    FailedToDecodeSignedTransaction,
+}
+
+impl From<RawEvmTxConversionError> for EthApiError {
+    fn from(e: RawEvmTxConversionError) -> Self {
+        match e {
+            RawEvmTxConversionError::EmptyRawTransactionData => {
+                EthApiError::EmptyRawTransactionData
+            }
+            RawEvmTxConversionError::FailedToDecodeSignedTransaction => {
+                EthApiError::FailedToDecodeSignedTransaction
+            }
+        }
+    }
+}
 
-impl TryFrom<RethBytes> for EvmTransaction {
-    type Error = EthApiError;
+impl TryFrom<RawEvmTransaction> for RethTransactionSignedNoHash {
+    type Error = RawEvmTxConversionError;
 
-    fn try_from(data: RethBytes) -> Result<Self, Self::Error> {
+    fn try_from(data: RawEvmTransaction) -> Result<Self, Self::Error> {
+        let data = RethBytes::from(data.tx);
         if data.is_empty() {
-            return Err(EthApiError::EmptyRawTransactionData);
+            return Err(RawEvmTxConversionError::EmptyRawTransactionData);
         }
 
-        let transaction = TransactionSigned::decode_enveloped(data)
-            .map_err(|_| EthApiError::FailedToDecodeSignedTransaction)?;
+        let transaction = RethTransactionSigned::decode_enveloped(data)
+            .map_err(|_| RawEvmTxConversionError::FailedToDecodeSignedTransaction)?;
 
-        let transaction = transaction
-            .into_ecrecovered()
-            .ok_or(EthApiError::InvalidTransactionSignature)?;
+        Ok(transaction.into())
+    }
+}
 
-        let (signed_transaction, signer) = transaction.to_components();
+impl TryFrom<RawEvmTransaction> for EvmTransactionSignedEcRecovered {
+    type Error = RawEvmTxConversionError;
 
-        let tx_hash = signed_transaction.hash();
-        let tx_eip_1559 = match signed_transaction.transaction {
-            reth_primitives::Transaction::Legacy(_) => {
-                return Err(EthApiError::InvalidTransaction(
-                    RpcInvalidTransactionError::TxTypeNotSupported,
-                ))
-            }
-            reth_primitives::Transaction::Eip2930(_) => {
-                return Err(EthApiError::InvalidTransaction(
-                    RpcInvalidTransactionError::TxTypeNotSupported,
-                ))
-            }
-            reth_primitives::Transaction::Eip1559(tx_eip_1559) => tx_eip_1559,
-        };
+    fn try_from(evm_tx: RawEvmTransaction) -> Result<Self, Self::Error> {
+        let tx = RethTransactionSignedNoHash::try_from(evm_tx)?;
+        let tx: RethTransactionSigned = tx.into();
+        let tx = tx
+            .into_ecrecovered()
```

Comment:
Here we reuse `ecrevovery` logic form `RethTransactionSigned` (imported form reth)